### PR TITLE
Gutenboarding: add PlansGrid accordion

### LINKED ISSUE #44428: Gutenboarding: Feature picker - Main list
## Design
Latest design iteration: 
- Feature picker: pcbO8q-2S-p2
- Plans grid: pcbO8q-bb-p2

## To do
- [ ] <s>Re-introduce [single column plans grid](https://github.com/Automattic/wp-calypso/pull/44299) reverted in #44582</s> Update Plans Grid to display recommended plans and selected features [Design in progress pcbO8q-bb-p2] #45227 
- [ ] [WIP] Save selected features in site options when creating a site in Gutenboarding and use the data to recommend correct plan during Complete Setup flow [@razvanpapadopol]
- [ ] Enable feature flag in production, add `?latest` query param to `/new` [redirect in signup controller for  newSiteGutenbergOnboarding test](https://github.com/Automattic/wp-calypso/blob/master/client/signup/controller.js#L132) and update A/B test slug. #45232

## Development

- [x] Add UI feature picker component. #44382 
- [x] Add single column, accordion styled plans grid. #44299
- [x] Display Feature picker as a step in site creation flow (hidden behind feature flag). #44382
- [x] Store selected features in the Onboard store and persist the data between sessions. #44431
- [x] Update Plans step with recommended plan based on selected features. When multiple feature are selected, pick the highest plan as `recommendedPlan`. #44431

## Follow-ups & bug fixing
- [x] Update copy. https://github.com/Automattic/wp-calypso/pull/44382#issuecomment-663461840 #44431
- [x] Fix visual bug when tabbing. https://github.com/Automattic/wp-calypso/pull/44382#pullrequestreview-454232301

### PR DESCRIPTION
**Pull request by razvanpapadopol**
#### Changes proposed in this Pull Request
* Re-add plans accordion in Gutenboarding and Site setup. **Note: PlansGrid accordion was first introduced in #44299**
* Ship Feature picker and Plans Grid accordion to Gutenboarding "stable" (still behind `gutenboarding/feature-picker` flag in production environment).
* Update Gutenboarding e2e test.

#### Testing instructions
* Go to [/new](https://hash-88877d0524e87950b8e6138c974bc58e78661c17.calypso.live/new)
* Before reaching Plans step in the linear flow, you should land on Feature selection step.
* After skipping or confirming feature selection, Plans step should display as an accordion.
* After creating the site, apply D48381-code and sandbox the new site.
* When in editor Site setup flow should include PlansGrid accordion.

### CODE REVIEW

**Comment by simison** [comment]
This still wouldn't launch it to customers, right? Either because we'd not be directing traffic to "experimental" flow yet or otherwise?

**Comment by razvanpapadopol** [comment]
> This still wouldn't launch it to customers, right? Either because we'd not be directing traffic to "experimental" flow yet or otherwise?

Yes, the experimental branch of the flow would be hidden behind `?latest` query param for both [Gutenboarding site creation](https://github.com/Automattic/wp-calypso/pull/45054/files#diff-ccb31dc04d72d8e153a03c27c2475568R101) and [Complete Setup](https://github.com/Automattic/wp-calypso/pull/45054/files#diff-0662eb41df2f7435f27f7f0dd064a7e9R67) flow.

**Comment by razvanpapadopol** [comment]
Superseded by #45280

**Comment by simison** [comment]
@razvanpapadopol there's at least one E2E failure from Gutenboarding that needs fixing.

File: client/landing/gutenboarding/stores/onboard/reducer.ts [hunk 1]
```diff
@@ -168,6 +155,19 @@ const selectedFeatures: Reducer< FeatureId[], OnboardAction > = (
 	return state;
 };
 
+const selectedFonts: Reducer< FontPair | undefined, OnboardAction > = (
```
Comment (simison) [inline client/landing/gutenboarding/stores/onboard/reducer.ts:158]: These were alphabetized in https://github.com/Automattic/wp-calypso/pull/45145 — let's keep them tidy. :-) 

Also nicer to review this PR without unrelated changes.

File: client/gutenberg/editor/calypsoify-iframe.tsx [hunk 2]
```diff
@@ -351,7 +351,7 @@ class CalypsoifyIframe extends Component<
 				this.props.siteCreationFlow === 'gutenboarding' && this.props.isSiteUnlaunched;
 			const frankenflowUrl = `${ window.location.origin }/start/new-launch?siteSlug=${ this.props.siteSlug }&source=editor`;
 			const isNewLaunch = config.isEnabled( 'gutenboarding/new-launch' );
-			const isExperimental = new URLSearchParams( window?.location.search ).has( 'latest' );
+			const isExperimental = config.isEnabled( 'gutenboarding/feature-picker' );
```
Comment (simison) [inline client/gutenberg/editor/calypsoify-iframe.tsx:354]: This "isExperimental" = "feature-picker" can feel a bit confusing later on but I think it's fine now, and we'll sort it out later.

Comment (razvanpapadopol) [reply client/gutenberg/editor/calypsoify-iframe.tsx:354]: Right now this would be easiest way to flip the switch between multi-column PlansGrid and accordion PlansGrid + Feature picker for both Gutenboarding and Site Setup flows.

Comment (simison) [reply client/gutenberg/editor/calypsoify-iframe.tsx:354]: Makes sense 👍

**Review by simison — COMMENTED** [review]
I'll test little more but two small visual findings. These can be follow-up PRs too:

Price loading state is a bit funky:
Screenshot 2020-09-10 at 11 36 16

Hiding the `/mo` completely vs placing it next to loading animation probably works best and is easiest.

Another is related to scroll positions:

- Expand all plans
- Click on domain to move to domains step
- Click continue
- Now you land in plans page where everything is collapsed again and scroll position isn't on top:

Screenshot 2020-09-10 at 11 37 45
Screenshot 2020-09-10 at 11 39 09
Screenshot 2020-09-10 at 11 39 13

- Easiest would be at least to ensure we're on top when landing on plans page
- It might be helpful to store expanded/collapsed state during the session

Again, neither should block merging.

**Review by simison — COMMENTED** [review]
In the launch flow, with flag enabled for a site that had free plan, I landed directly at plan step:

image

I expected to land on domain step first like I do in `master`.

Without the flag, I see older "launch" button instead of "complete setup" button. I expected complete setup with multi-grid:

image

**Comment by razvanpapadopol** [comment]
> Price loading state is a bit funky:

Fixed in https://github.com/Automattic/wp-calypso/pull/45054/commits/a659c2e8cb727a6dcabd52a6c97428cb30b3a055
Demo on mobile: https://cloudup.com/c4NHHotDhJ7
Demo on desktop: https://cloudup.com/cOd1LafAdQG

> Another is related to scroll positions:
> * Easiest would be at least to ensure we're on top when landing on plans page

Fixed in https://github.com/Automattic/wp-calypso/pull/45054/commits/88877d0524e87950b8e6138c974bc58e78661c17

**Review by simison — APPROVED** [review]
Double tested with new site with Chrome; looks like the landing on site title step was caused by something else than this PR (happens also in `master`).

Other than that, the flow is looking good and works great. 👍  Let's get this in.
